### DISCUSSION TITLE
Games and Music

### DISCUSSION THREAD

**Discussion by electricityforprogress** [joeycastillo/Sensor-Watch#41] [General]
I am thinking about making some sort of game(s) which use the LCD display.  Most likely they would be 'controllable animations'.  I was also considering a simple 'music sequencer' which could play 1bit ringtone style sounds, you know with a 303 style step sequencer ;)

I will do a couple drawings of the LCD and different combinations of segments which might make for a simple game face.

**Comment by electricityforprogress** [comment]
I've done some research on the old 'Digital Invaders' game which is a numbers matching game with moving strings of numbers, controls are 'aim' and 'fire' buttons, as seen on teh CA-503 and some calculators.  I am also trying to dream up some ideas for a side scroller adventure game, perhaps the google dinosaur jumping game, and of course some face of animations.   DnD dice roller would be fun!  

Please jump in with any ideas for a simple game or any recommendations for old games to try!

https://www.youtube.com/watch?v=UIZgzbtg5j8

**Comment by jdmoore0883** [comment]
I've been thinking about what kind of games could fit within the display constraints (this is the tougher part). I have come up with the following:

- Nim (https://en.wikipedia.org/wiki/Nim) (https://rosettacode.org/wiki/Nim_game). Obviously we wouldn't want to make the game unbeatable like that (maybe a "joke" game against your friends ;) ), but one can take inspiration from the code examples. 
- Blackjack / 21. We'd have trouble identifying the suit, but it also isn't really necessary.
- Baccarat (https://en.wikipedia.org/wiki/Baccarat).
- Could maybe even combine the 2 (and maybe some others) in to a sort of "casino" game(s).

Those were some ideas for some on-watch-only games. But I had a thought... using an "external"/"real" chess board and chess notations (https://en.wikipedia.org/wiki/Algebraic_notation_(chess)), one could turn the watch in to a chess game/opponent/computer. Enter your move, the watch calculates it's move and displays the notation for it's move.

**Reply by electricityforprogress** [reply]
Oh yeah, chess is a great idea, a real board in the real world and notation on the watch!

**Comment by HiAaronCody** [comment]
My first stop would be a simple dice or coin-flip app. Useful for lots of games outside the watch.

**Reply by electricityforprogress** [reply]
A random coin or dice would fit the display pretty well. I was thinking about a DnD dice roller, the display has enough characters to display the dXX and show two rolls for advantage/disadvantage.  

you are right lots of real world dice games could be driven off the watch roller.  Makes me want one player yahtzee ;)

**Comment by niclashoyer** [comment]
As for music, I just played around with the `buzzer-test` app and created a small script that can convert notes from here:

https://github.com/robsoncouto/arduino-songs

So, we can at least enjoy tetris music until we find out how to display tetris on the display 😆 

run this in `apps/buzzer-test` and paste notes from e.g. https://github.com/robsoncouto/arduino-songs/blob/master/tetris/tetris.ino#L114-L142

https://gist.github.com/niclashoyer/d65125780c446d0e374c1bde82ff6fed

**Comment by ericlimer** [comment]
A good avenue for game development, IMO, is to try and expand on the traditional reflex/time-estimation "stopwatch games" of: 

- trying to start and stop the stopwatch as quickly as possible
- trying to stop the stopwatch on exactly 5.00 or 10.00
- trying to stop the stopwatch on exactly 5.00 or 10.00 without looking at it

By adding streak mechanics or cumulative scoring systems and abstracting the countdown aspect of it, you could probably make some decently compelling games! A few off the top of my head: 

- Golf: The watch runs through a series of countdowns from 1 second through 10 seconds, in random order (1-18 if you wanna turn up the golf dial). The player attempts to press a button at exactly 00.00. The countdown goes into the negative or reascends if they're late. At the end of the series, the score is the cumulative time that the player was off from zero across all "holes." (Zero is a perfect score). With countdowns visible, it's a reflex game. With countdowns hidden, it's a time-estimation game. You could even mess around with just hiding parts of the countdown, like showing the full seconds but hiding the tenths and hundredths. 
- Racing: One lit segment takes a route around the screen (could be an oval, or something more complex; maybe there are even multiple courses) for a set number of laps. The player has to hit a button while the segment is in a specific section. If the player succeeds, the speed increases. If the player presses at the wrong time, the speed decreases. If the player does nothing, the speed stays the same. The result is a reflex game with a sort of betting mechanic and a (ideally near-impossible) perfect score to measure progress against.
